cmake/find/FindJulia.cmake

```diff
@@ -0,0 +1,108 @@
+macro(julia_bail_if_false message var)
+  if(NOT ${var})
+    set(Julia_FOUND 0)
+    set(Julia_NOT_FOUND_MESSAGE "${message}")
+    return()
+  endif()
+endmacro()
+
+# https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_BINDIR
+find_program(JULIA_EXECUTABLE julia PATHS ENV JULIA_BINDIR)
+julia_bail_if_false("The Julia executable could not be found" JULIA_EXECUTABLE)
+
+if(NOT DEFINED JULIA_BINDIR)
+  # The executable could be a chocolatey shim, so run some Julia code to report
+  # the path of the BINDIR
+  execute_process(
+      COMMAND "${JULIA_EXECUTABLE}" -e "print(Sys.BINDIR)"
+      OUTPUT_VARIABLE julia_bindir
+  )
+  file(TO_CMAKE_PATH "${julia_bindir}" julia_bindir)
+  set(JULIA_BINDIR "${julia_bindir}" CACHE PATH "")
+endif()
+julia_bail_if_false("The Julia executable could not report its location" JULIA_BINDIR)
+
+get_filename_component(julia_prefix "${JULIA_BINDIR}" DIRECTORY)
+
+if(WIN32)
+  set(julia_old_CMAKE_FIND_LIBRARY_SUFFIXES "")
+  set(julia_old_CMAKE_FIND_LIBRARY_PREFIXES "")
+  if(CMAKE_FIND_LIBRARY_SUFFIXES)
+    set(julia_old_CMAKE_FIND_LIBRARY_SUFFIXES "${CMAKE_FIND_LIBRARY_SUFFIXES}")
+    list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .dll.a)
+  else()
+    set(CMAKE_FIND_LIBRARY_SUFFIXES ".lib;.dll.a")
+  endif()
+  if(CMAKE_FIND_LIBRARY_PREFIXES)
+    set(julia_old_CMAKE_FIND_LIBRARY_PREFIXES "${CMAKE_FIND_LIBRARY_PREFIXES}")
+    list(APPEND CMAKE_FIND_LIBRARY_PREFIXES lib)
+  else()
+    set(CMAKE_FIND_LIBRARY_PREFIXES ";lib")
+  endif()
+endif()
+
+find_library(JULIA_LIBRARY julia HINTS "${julia_prefix}/lib")
+
+if(WIN32)
+  set(CMAKE_FIND_LIBRARY_SUFFIXES "${julia_old_CMAKE_FIND_LIBRARY_SUFFIXES}")
+  set(CMAKE_FIND_LIBRARY_PREFIXES "${julia_old_CMAKE_FIND_LIBRARY_PREFIXES}")
+endif()
+
+julia_bail_if_false("The Julia library could not be found" JULIA_LIBRARY)
+
+find_path(
+    JULIA_INCLUDE_DIR julia.h
+    HINTS "${julia_prefix}/include" "${julia_prefix}/include/julia"
+)
+julia_bail_if_false("The Julia header could not be found" JULIA_INCLUDE_DIR)
+
+if(NOT DEFINED JULIA_VERSION)
+  file(STRINGS "${JULIA_INCLUDE_DIR}/julia_version.h" julia_version LIMIT_COUNT 1 REGEX JULIA_VERSION_STRING)
+  string(REGEX REPLACE ".*\"([^\"]+)\".*" "\\1" julia_version "${julia_version}")
+  set(JULIA_VERSION "${julia_version}" CACHE STRING "Version of Julia")
+endif()
+
```

Comment:
```suggestion
if (NOT ${JULIA_VERSION} STRGREATER_EQUAL "1.7.0")
    message(WARNING "Julia version 1.7.0 or newer is required, but version ${JULIA_VERSION} is installed. You can download the latest version at: https://julialang.org/downloads/ ")
endif()
```

Reply:
While this would be better done using cmake, this way we can link to the downloads page for julia manually

Reply:
Personally, I didn't even think of looking at Julia's homepage, I just ran `choco install julia`. CMake will already error out if the version is not sufficiently new enough.

Reply:
If that already works then we don't need this error